hack/verify-go-modules.sh

```diff
@@ -55,42 +86,40 @@ function diff_version_deps {
   has_deps="${1}"
   wants_deps="${2}"
   jq -s '
-    # Extracts v<Maj>.<Min> from a semantic version string.
-    def major_minor_semver: capture("v(?<major>\\d+)\\.(?<minor>\\d+)")
-      | "v" + .major + "." + .minor;
-
     map(to_entries)    # Convert both input dicts into two separate arrays.
       | add            # Concatenate those two arrays into a single one.
       | group_by(.key) # Group items by `.key`.
       | map(           # Map each selected item into { "<Dep>": {"has": "<Version0>", "wants": "<Version1>"} }.
           select(
               # If grouping resulted in two items, it means both arrays have this dependency.
               length == 2
-              # Compare the v<Maj>.<Min> of both items.
-              and (.[0].value | major_minor_semver) != (.[1].value | major_minor_semver)
+              # And the dependency has a version mismatch.
+              and (.[0].value  != .[1].value)
             )
             | { (.[0].key): { "has": .[0].value, "wants": .[1].value } }
         )
       | add // empty
   ' "${has_deps}" "${wants_deps}"
 }
 
-# print_diff_version_deps prints the output of diff_version_deps as a
-# human-readable multi-line text.
+# print_diff_version_deps prints the output of diff_version_deps (expected in stdin)
+# as a human-readable multi-line text.
+# Returns 1 if any lines were printed (i.e. errors were found).
 function print_diff_version_deps {
-  jq -r "to_entries | map(\"Warning: version mismatch: has \(.key)@\(.value.has), but \(.value.wants) expected\") | join(\"\\n\")"
+  jq -r '
+    to_entries
+      | map("Version mismatch: \(.key) is \(.value.has), but \(.value.wants) expected")
```

Comment:
```suggestion
      | map("  Version mismatch: \(.key) is \(.value.has), but \(.value.wants) expected")
```

Just as a visual separator in the output - I felt like the mismatches were going missing a bit in the output